lib/Operators.ml

```diff
@@ -4,37 +4,39 @@ open Core_kernel
 open Type_conversion
 open Ast
 
+let ternary_if = "TernaryIf__"
+
 (** A hash table to hold some name conversions between the AST nodes and the
     Stan Math name of the operator *)
 let operator_names = String.Table.create ()
 
-let _ = Hashtbl.add_multi operator_names ~key:"Plus" ~data:"add"
+let _ = Hashtbl.add_multi operator_names ~key:"Plus__" ~data:"add"
 
 (*XXX is there a difference in Stan math for unary plus?*)
-let _ = Hashtbl.add_multi operator_names ~key:"PPlus" ~data:"plus"
-let _ = Hashtbl.add_multi operator_names ~key:"Minus" ~data:"subtract"
+let _ = Hashtbl.add_multi operator_names ~key:"PPlus__" ~data:"plus"
+let _ = Hashtbl.add_multi operator_names ~key:"Minus__" ~data:"subtract"
 
 (*XXX is there a difference in Stan math for unary minus?*)
-let _ = Hashtbl.add_multi operator_names ~key:"PMinus" ~data:"minus"
-let _ = Hashtbl.add_multi operator_names ~key:"Times" ~data:"multiply"
-let _ = Hashtbl.add_multi operator_names ~key:"Divide" ~data:"divide"
-let _ = Hashtbl.add_multi operator_names ~key:"Divide" ~data:"mdivide_right"
-let _ = Hashtbl.add_multi operator_names ~key:"Modulo" ~data:"modulus"
-let _ = Hashtbl.add_multi operator_names ~key:"LDivide" ~data:"mdivide_left"
-let _ = Hashtbl.add_multi operator_names ~key:"EltTimes" ~data:"elt_multiply"
-let _ = Hashtbl.add_multi operator_names ~key:"EltDivide" ~data:"elt_divide"
-let _ = Hashtbl.add_multi operator_names ~key:"Pow" ~data:"pow"
-let _ = Hashtbl.add_multi operator_names ~key:"Or" ~data:"logical_or"
-let _ = Hashtbl.add_multi operator_names ~key:"And" ~data:"logical_and"
-let _ = Hashtbl.add_multi operator_names ~key:"Equals" ~data:"logical_eq"
-let _ = Hashtbl.add_multi operator_names ~key:"NEquals" ~data:"logical_neq"
-let _ = Hashtbl.add_multi operator_names ~key:"Less" ~data:"logical_lt"
-let _ = Hashtbl.add_multi operator_names ~key:"Leq" ~data:"logical_lte"
-let _ = Hashtbl.add_multi operator_names ~key:"Greater" ~data:"logical_gt"
-let _ = Hashtbl.add_multi operator_names ~key:"Geq" ~data:"logical_gte"
-let _ = Hashtbl.add_multi operator_names ~key:"PNot" ~data:"logical_negation"
-let _ = Hashtbl.add_multi operator_names ~key:"Transpose" ~data:"transpose"
-let _ = Hashtbl.add_multi operator_names ~key:"TernaryIf" ~data:"if_else"
+let _ = Hashtbl.add_multi operator_names ~key:"PMinus__" ~data:"minus"
```

Comment:
Hmm, I was expecting the rhs to have the underscores here and to be used as the name for code gen. (It would also need to be changed in Stan_math_signatures then.) I need to think about this a bit more.

Reply:
if it was on the RHS or just in code gen, we wouldn't be able to tell between our operator and a user calling their own "Plus" function in the MIR.